core/controllers/collection_editor.py

```diff
@@ -155,40 +156,60 @@ def get(self, collection_id):
 
         self.render_json(self.values)
 
-    @acl_decorators.can_manage_collection_publish_status
+
+class CollectionPublishHandler(base.BaseHandler):
+
+    @acl_decorators.can_publish_collection
+    def put(self, collection_id):
+        """Publishes the given collection."""
+        collection = collection_services.get_collection_by_id(collection_id)
+        version = self.payload.get('version')
+        _require_valid_version(version, collection.version)
+        collection_rights = rights_manager.get_collection_rights(
+            collection_id, strict=False)
+
+        try:
+            collection.validate(strict=True)
+            collection_services.validate_exps_in_collection_are_public(
+                collection)
+        except utils.ValidationError as e:
+            raise self.InvalidInputException(e)
+
+        collection_services.publish_collection_and_update_user_profiles(
+            self.user, collection_id)
+        collection_services.index_collections_given_ids([
+            collection_id])
+
+        self.values.update({
+            'can_edit': True,
+            'can_unpublish': rights_manager.check_can_unpublish_activity(
```

Comment:
Between retrieving the rights and checking them here, you've published the collection. So the status field would have changed. Is that an issue? Ditto for rights_manager stuff below.